pandas/tests/io/json/test_pandas.py

```diff
@@ -1032,6 +1038,102 @@ def test_to_jsonl(self):
         assert result == expected
         assert_frame_equal(pd.read_json(result, lines=True), df)
 
+    def test_readjson_chunks(self, lines_json_df):
+        """Basic test that read_json(chunks=True) gives the same result as
+        read_json(chunks=False)"""
+        # GH17048: memory usage when lines=True
+
+        for cs in [1, 1.0]:
+
+            unchunked = pd.read_json(StringIO(lines_json_df), lines=True)
+            chunked = pd.concat(
+                pd.read_json(StringIO(lines_json_df), lines=True, chunksize=cs)
+            )
+
+            assert_frame_equal(chunked, unchunked)
+
+    def test_readjson_chunksize_requires_lines(self, lines_json_df):
+        msg = "chunksize can only be passed if lines=True"
+        with tm.assert_raises_regex(ValueError, msg):
+            pd.read_json(StringIO(lines_json_df), lines=False, chunksize=2)
+
+    def test_readjson_chunks_series(self):
+        """Test reading line-format JSON to Series with chunksize param"""
+        s = pd.Series({'A': 1, 'B': 2})
+
+        strio = StringIO(s.to_json(lines=True, orient="records"))
+        unchunked = pd.read_json(strio, lines=True, typ='Series')
+
+        strio = StringIO(s.to_json(lines=True, orient="records"))
+        chunked = pd.concat(pd.read_json(
+            strio, lines=True, typ='Series', chunksize=1
+        ))
+
+        assert_series_equal(chunked, unchunked)
+
+    def test_readjson_each_chunk(self, lines_json_df):
+        """
+        Other tests check that the final result of read_json(chunksize=True) is
+        correct. This checks that the intermediate chunks read in are correct.
+        """
+        chunks = list(
+            pd.read_json(StringIO(lines_json_df), lines=True, chunksize=2)
+        )
+        assert chunks[0].shape == (2, 2)
+        assert chunks[1].shape == (1, 2)
+
+    def test_readjson_chunks_from_file(self):
+        with ensure_clean('test.json') as path:
+            df = pd.DataFrame({'A': [1, 2, 3], 'B': [4, 5, 6]})
+            df.to_json(path, lines=True, orient="records")
+            chunked = pd.concat(pd.read_json(path, lines=True, chunksize=1))
+            unchunked = pd.read_json(path, lines=True)
+            assert_frame_equal(unchunked, chunked)
+
+    def test_readjson_chunks_closes(self):
+        for chunksize in [None, 1]:
+            with ensure_clean('test.json') as path:
+                df = pd.DataFrame({'A': [1, 2, 3], 'B': [4, 5, 6]})
+                df.to_json(path, lines=True, orient="records")
+                f = open(path, 'r')
+                if chunksize is not None:
+                    pd.concat(pd.read_json(f, lines=True, chunksize=chunksize))
+                else:
+                    pd.read_json(f, lines=True)
```

Comment:
this is not correct
we should only close the file if we open it ourselves
if we have an open file it should stay open

Reply:
Ah, that makes sense and is in line with current behavior. So in [this comment](https://github.com/pandas-dev/pandas/pull/17168/files#r139048920), did you mean something like the following?

```
           df = pd.DataFrame({'A': [1, 2, 3], 'B': [4, 5, 6]})                      
            df.to_json(path, lines=True, orient="records")                           
            reader = JsonReader(                                                     
                path, orient=None, typ="frame", dtype=True, convert_axes=True,   
                convert_dates=True, keep_default_dates=True, numpy=False,        
                precise_float=False, date_unit=None, encoding=None,              
                lines=True, chunksize=chunksize)                                 
            reader.read()                                                        
            assert reader.open_stream.closed
```